### DISCUSSION TITLE
how can I know which type of errors the "wr.opensearch.delete_index" can throw ?

### DISCUSSION THREAD

**Discussion by arielvol** [aws/aws-sdk-pandas#1217] [Q&A]
I want to create a method that receives a list of indexes name and then, in a loop, calls the "wr.opensearch.delete_index" method and deletes them one by one.

However, I don't want to fail the all deletion if there is an issue with a specific delete, so I want to use a try - except expression.

I usually try to avoid a generic "except" phrase and try to catch the exact error, but I can't find any documentation that can tell me which type of errors I might get from the "wr.opensearch.delete_index" method call.

**Comment by jaidisido** [comment]
Hi @arielvol, the [method](https://github.com/awslabs/aws-data-wrangler/blob/main/awswrangler/opensearch/_write.py#L244) is simply issuing a `delete_index` opensearch API [call](https://opensearch.org/docs/1.2/opensearch/rest-api/index-apis/delete-index). Note that 400 and 404 errors (i.e. index not found) are ignored by default.

A list of exceptions can be found here: https://github.com/opensearch-project/opensearch-py/blob/main/opensearchpy/exceptions.py
